feat(docs): update forum moderation section

hbar1st opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->
Updates to the section on forum moderation in the moderator guidebook. (including clarification on spam handling and using cellphone to perform mod actions)

hbar1st committed:
Update moderator-handbook.md

- add a section describing some of the quirks of moderating with a cell-phone

staranbeer committed:
updated the "dealing with spam" section

staranbeer committed:
added  a "posted solution" section to mod-guide

staranbeer committed:
minor edits in "dealing with solutions" section of the mod guide

hbar1st committed:
Update moderator-handbook.md

Added details for contacting support with required details when user is underage. Started a section on dealing with forum flags but it is incomplete until more feedback is received from the mod-team.

staranbeer committed:
updated the wording for the solutions sections (mod-guide update)

staranbeer committed:
added dealing with flags draft (mod-guide update)

hbar1st committed:
Refinements to handling spam section.

hbar1st committed:
Undo of flag handling section

hbar1st committed:
Merge pull request #1 from freeCodeCamp/main

pr test

hbar1st committed:
Merge branch 'main' into main

ShaunSHamilton commented on docs/moderator-handbook.md:
```diff
@@ -171,11 +171,17 @@ Forum moderators can delete users' posts. You should only do this for the follow
+
+#### Some notable situations:
+
+1. FreeCodeCamp's spam filter requires a user to take a certain amount of time to ask their question. If they click on the "Create post" button before that, they or their post might get flagged for review for being potential spam. In that case, it is the responsibility of the moderators to approve their post if they are a genuine user and need help. The same applies if the post is flagged as spam ...
```
I am not sure why this needs to be mentioned. All review requests fall under the moderators' responsibility to review.

```suggestion
1.  The _"min first post typing time"_ spam filter requires a low-level user to take a certain amount of time to create a post. If they click on the "Create post" button before that, their post might get flagged for review, and their account will get a note.
```

ShaunSHamilton commented on docs/moderator-handbook.md:
```diff
@@ -171,11 +171,17 @@ Forum moderators can delete users' posts. You should only do this for the follow
-For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin panel). Send the user a warning with the [Code of Conduct](https://code-of-conduct.freecodeca ...
+For the first spam post of a legitimate user (ie. whose intent isn't to spam the forum but to learn/contribute to the forum), send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin pa ...
+
+#### Some notable situations:
```
My main suggestion is to remove this section.

```suggestion
#### Notable Situations:
```

hbar1st replied:
I was hoping to add some information here to explicitly call out responsibilities that new mods need to be aware of. (I think for the newest mods, we all share in some confusion of how to handle requests for review, it took a few tries at least and discussion to get insight from other mods on this - frankly there are still some situations that I'm not clear on, so yes, this section is not ideal as-is for what I was hoping for, but we need something to direct new mods to these notable situations (akismet flags, user flags, tricky spammers, etc).

The amount of spam I've clicked on in my first 2 weeks has made me very uncomfortable (because I'm essentially clicking blind to see if this is spam or something else). Calling out the behaviour of adding links after the post is up would have helped me to identify the spammer without having to click a link to try to understand the objective of the post.

Having said that, I'm still not happy with this section. We can defer the update till the new mods and the more experienced mods can agree on what new mods need to know.

ShaunSHamilton replied:
I hope my suggestions do not come across as harsh; I think updating the mod handbook is an excellent idea.

I think we might need to tackle this from a different standpoint. That is, I suggest just _updating_ the existing sections here, and add a completely new section aimed at _onboarding_ new mods.

Specifically, I would suggest:
1) We should double check the existing docs for out-dated info. Then, remove that info. **No major additions of text**
2) Create a new high-level section `### Onboarding Information`
  - `#### How to Use the Forum` - where to find stuff new (visible) to mods
  - `#### Responsibilities and Expectations`
  - `#### Common Patterns and Actions` - what to expect from bots, spammers, and bad-actors

Ideally, we have:
- _Mod Code of Conduct_ - a guide that tells step-by-step for common situations
- Something less official where most mods will learn by watching. E.g. how to handle solution posters, how to help with support requests, how to handle mis-use of the flag system, how to use canned replies*

\* we should not have a "canned replies" section in the mod book, because this does change. We only have 2/3 canned replies that are "official" (always used in the same situation, following moderation action). Otherwise, most of the canned replies are just "nice to haves" - new mods should take a couple minutes familiarising themselves with what currently exists.

---

That being said, let us wait to hear what other mods think, before putting in the work to adjust this.

ShaunSHamilton replied:
> The amount of spam I've clicked on in my first 2 weeks has made me very uncomfortable (because I'm essentially clicking blind to see if this is spam or something else). 

For the most part, this is something you end up learning. In the last 2 years, I cannot remember clicking on a single link to "check" if it is spam or not. We are a programming forum - there are only so many domains that are on topic.
There is no requirement for any mod to put themselves in a position they might be uncomfortable with. So, if you are unsure about a link, feel free to flag/reach-out before clicking.

Otherwise, something you can do: Use a different browser/profile with the strictest settings (block images, prevent JavaScript, block all cookies, etc.) to open unknown links.

SaintPeter replied:
Just a grammar/style note - if you have header markup, you don't need a trailing colon. It also doesn't match the other headers which do not have them.

ShaunSHamilton commented on docs/moderator-handbook.md:
```diff
@@ -171,11 +171,17 @@ Forum moderators can delete users' posts. You should only do this for the follow
+
+1. FreeCodeCamp's spam filter requires a user to take a certain amount of time to ask their question. If they click on the "Create post" button before that, they or their post might get flagged for review for being potential spam. In that case, it is the responsibility of the moderators to approve their post if they are a genuine user and need help. The same applies if the post is flagged as spam ...
+
+2. If the user is an obvious spam bot, for example, their first and only post is a link to a malicious/self-promoting site minutes after signing up, delete their account.
```
This is unnecessary - it is covered by the third point in _Dealing with Spam_

hbar1st replied:
i think we were trying to mention some of the tactics that spammers user but I see we didn't quite make the tactics clear here.

naomi-lgbt replied:
I'm not keen on listing out the specific qualifiers for spam. Spam is either blatantly obvious, or it's something that we end up flagging and discussing.

If we create a list, we'll start seeing people challenge decisions because it doesn't match the list (s'why our code of conduct is so concise).

I'd rather encourage moderators to flag as "Something Else" and start a discussion thread than try to maintain a definition of spam.

ShaunSHamilton commented on docs/moderator-handbook.md:
```diff
@@ -187,11 +193,15 @@ In exceptional circumstances, it may be appropriate for a moderator to fork a di
 
 Again, if you have any problems or questions, make a post with your actions in the `"Staff"` category, and tag another moderator if you want them to review your moderating actions.
 
+### Dealing with Posted Solutions
+
+When a user posts a solution to any of the questions, remove it and either, kindly let them know that solutions are not permitted on the forum in your own words or, use the **Solution Instead of Help** "Canned reply".
+
```
Personally, I would not add this to a moderator handbook.

It is not spammy to post a solution (once). We will just never encourage the sharing/posting of solutions, because it rarely aids anyone's learning.

Essentially, it is not a "must remove solution" rule; it is a "encourage proactive learning" rule.

hbar1st replied:
yes true.  I think we can rewrite to:

When a user responds to another user's post by posting a solution, remove the solution and suggest that tips/hints are a better way to help others learn (alt. use the **Solution Instead of Help** "Canned reply").

naomi-lgbt replied:
> Essentially, it is not a "must remove solution" rule; it is a "encourage proactive learning" rule.

The practise lately, AFAIK, has been to remove/edit out a solution in a post if that solution was provided in response to a camper's request for help, but that a solution being shared for the sake of discussing a solution is fine.

ShaunSHamilton commented on docs/moderator-handbook.md:
```diff
@@ -187,11 +193,15 @@ In exceptional circumstances, it may be appropriate for a moderator to fork a di
 Our [Terms of Service](https://freecodecamp.org/terms) require that freeCodeCamp users be at least 13 years of age. If a user reveals that they are under the age of 13, send them the below message and delete their forum account (if deletion is not available, suspending the account is sufficient).
 
-**Email `support[at]freecodecamp.org` to delete the user's freeCodeCamp account as well.**
+**Email `support[at]freecodecamp.org` to delete the user's freeCodeCamp account as well (providing the email address and user id to support).**
```
The user id is generated by Discourse, and is in no way connected to the `/learn` account. So, it is often more useful to provide the username which might match the `/learn` username.

```suggestion
**Email `support[at]freecodecamp.org` to delete the user's freeCodeCamp account as well (providing the email address and username of the offending forum account).**
```

ShaunSHamilton commented on docs/moderator-handbook.md:
```diff
@@ -171,11 +171,17 @@ Forum moderators can delete users' posts. You should only do this for the follow
 1. Someone has posted a pornographic or graphically violent image.
 2. Someone has posted a link or code that is malicious in nature and could harm other campers who click on it.
-3. Someone has flooded a thread with a lot of spam messages.
+3. Someone has flooded a thread with a lot of spam messages or is an obvious spam-bot or spammer from the outset (as determined by the pattern of posting once for example then adding a spam link to their post later on to try to defeat the automated spam detection).
```
```suggestion
3. Someone has flooded a thread with a lot of spam messages.
4. An account has been created, beyond reasonable doubt, to spam.
```

hbar1st committed:
chg user id to username in under-13 mod guide section

Co-authored-by: Shaun Hamilton <shauhami020@gmail.com>

hbar1st committed:
add 4th point to include obvious spammers

Co-authored-by: Shaun Hamilton <shauhami020@gmail.com>

SaintPeter commented on docs/moderator-handbook.md:
```diff
@@ -172,10 +172,17 @@ Forum moderators can delete users' posts. You should only do this for the follow
 1. Someone has posted a pornographic or graphically violent image.
 2. Someone has posted a link or code that is malicious in nature and could harm other campers who click on it.
 3. Someone has flooded a thread with a lot of spam messages.
+4. An account has been created, beyond reasonable doubt, to spam.
```
```suggestion
4. An account has been created, beyond a reasonable doubt, to spam.
```

jeremylt commented on docs/moderator-handbook.md:
```diff
@@ -172,10 +172,17 @@ Forum moderators can delete users' posts. You should only do this for the follow
 ### Dealing with Spam
 
-For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin panel). Send the user a warning with the [Code of Conduct](https://code-of-conduct.freecodeca ...
+For the first spam post of a legitimate user (ie. whose intent isn't to spam the forum but to learn/contribute to the forum), send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin pa ...
```
```suggestion
For the first spam post of a legitimate user (ie. whose intent isn't to spam the forum but to learn/contribute to the forum), send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then silence the user. Send the user a warning with the [Code of Conduct](https://code-of-conduct.freecodecamp.org). Check the box in the private message indicating that your message is a "formal warning."
```
minor simplification?

jeremylt commented on docs/moderator-handbook.md:
```diff
@@ -187,11 +194,15 @@ In exceptional circumstances, it may be appropriate for a moderator to fork a di
 Our [Terms of Service](https://freecodecamp.org/terms) require that freeCodeCamp users be at least 13 years of age. If a user reveals that they are under the age of 13, send them the below message and delete their forum account (if deletion is not available, suspending the account is sufficient).
 
-**Email `support[at]freecodecamp.org` to delete the user's freeCodeCamp account as well.**
+**Email `support[at]freecodecamp.org` to delete the user's freeCodeCamp account as well (providing the email address and username of the offending forum account).**
```
might be worth noting in this section that only admins can access the email addresses of forum users

naomi-lgbt replied:
Honestly, I think this flow should be updated entirely.

A moderator should suspend the forum account, send the following message, and then ping a member of staff to delete the accounts - this way we can grab the email address to check for a `/learn` account.

hbar1st replied:
Hi Naomi, I think the guide basically says the same thing (but in a slightly different order)?

naomi-lgbt reviewed (commented):
I've left some thoughts in the existing conversation threads.

I have thoughts on the onboarding document, but since that's out of scope for the PR I'll hit the [forum post](https://forum.freecodecamp.org/t/mod-guide-new-section-onboarding-new-mods/555304).

hbar1st committed:
Undo Notable Situations addition

Co-authored-by: Shaun Hamilton <shauhami020@gmail.com>

hbar1st committed:
Update docs/moderator-handbook.md

Co-authored-by: Rex Schrader <rex.schrader@gmail.com>

hbar1st committed:
Update to min first post typing paragraph

Co-authored-by: Shaun Hamilton <shauhami020@gmail.com>

SaintPeter commented on docs/moderator-handbook.md:
```diff
@@ -172,10 +172,17 @@ Forum moderators can delete users' posts. You should only do this for the follow
-For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin panel). Send the user a warning with the [Code of Conduct](https://code-of-conduct.freecodeca ...
+For the first spam post of a legitimate user (ie. whose intent isn't to spam the forum but to learn/contribute to the forum), send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin pa ...
+
+#### Notable Situations:
```
```suggestion
#### Notable Situations
```
Remove the colon as it is unneeded and is different from other headers.

hbar1st committed:
Update docs/moderator-handbook.md

Co-authored-by: Rex Schrader <rex.schrader@gmail.com>

hbar1st committed:
re-order underage forum user steps

SaintPeter reviewed (dismissed):
LGTM 🎉

hbar1st committed:
removed notable situations 

we will work on a future doc for moderator onboarding to mention notable situations

SaintPeter reviewed (dismissed):
(Still) LGTM!  🎉 🌹

naomi-lgbt reviewed (dismissed):
Let's go ahead and get this in.

Sboonny commented on docs/moderator-handbook.md:
```diff
@@ -203,6 +207,13 @@ Please rejoin once you have reached at least 13 years of age.
 
+### Moderating Via Cell-phone
+
+Moderating the forum is possible via a cell-phone but you may encounter some usage quirks. This is not an exhaustive list.
```
```suggestion
Moderating the forum is possible via a cell-phone but you may encounter some usage quirks. This is not an exhaustive list.

```

hbar1st committed:
Fix Linting Problem
